doc/_downloads/modi_veros_run.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash -l
+#
+# Defines where the package should be installed.
+# Since the ~/modi_mount directory content is
+# available on each node, we define the package(s) to be installed
+# here so that the node can find it once the job is being executed.
+export CONDA_PKGS_DIRS=~/modi_mount/conda_dir
+
+# Activate conda in your $PATH
+# This ensures that we discover every conda environment
+# before we try to activate it.
+source $CONDA_DIR/etc/profile.d/conda.sh
+
+# Check if the directory with package(s) exists, otherwise create it
+if [ ! -d $CONDA_PKGS_DIRS ]; then
+        mkdir $CONDA_PKGS_DIRS
+fi
+
+# As per https://veros.readthedocs.io/en/latest/introduction/get-started.html#installation
+# we download and install the Veros environment unless it is already available
+if [ ! -d "veros" ]
+then
+      git clone https://github.com/team-ocean/veros.git -b v0.2.3
+else
+      echo "Veros source directory exists"
+      echo "Make sure it contains the rigth model version!"
+fi
+
+# Either activate the existing veros environment or create a new one
+conda activate veros
+if [ $? != 0 ]; then
+      conda create -n veros -y python=3.7
+      conda activate veros
+
+      # Install the packages into the conda environment that was activated.
+      pip3 install ./veros
+      pip3 install veros-bgc
+fi
```

Comment:
I just followed MODI guideline. But I agree, it must be possible to install VerosB manually. Let me test the manual way and if that works, I will add it to MODI installation instructions.

Reply:
It turned out to be impossible. 🤷‍♂️ The MODI system is built in a way that software on a login node doesn't talk to software on a compute node. So, it means that one must install the software (if that is not available in a singularity image) via batch script in $HOME/.conda/envs/... directory on a compute node in order to use it there.

Reply:
Huh, that's too bad. That means the file system is completely separate? So you can't even edit the setup file on the login node?

Reply:
They are not completely separate. You can edit setup, but it must be in ~/erda_mount. It is the only shared disk space between login and compute nodes. 
But you cannot check what packages have been already installed (available). For that you need to submit a Slurm job (batch script), which runs a script having this command line:
`conda list`  🤪 

Or you cannot copy a setup from the setup gallery on the login node because Veros is installed on the compute nodes.

In lite of this situation, I guess it is ok to keep the scripts as they are.

I will modify the step 8 for DAG (below) and I guess we can merge this branch to the master...

Reply:
What if we supply a script `modi_veros_install.sh` that contains essentially the first half of this script and that you can run on the login node first (which allows you to use `veros copy-setup` and populates the `CONDA_PKGS_DIRS` cache)? 

My main gripe with the current solution is this:

```bash
$ sed -i 's/vs.runlen = 0/vs.runlen = 86400.0 * 360 * 10/g' bgc_global_four_degree.py
```

As soon as people want to make non-trivial changes to the setup they will hit a brick wall, so if we can avoid it it would be a huge step forward.

Reply:
I see! 
Do you thing having two scripts like
modi_veros_setup.sh (with the code before `$ sed -i ...` )
modi_veros_run.sh (with `python3 bgc_global_four_degree.py` only)
will be a good idea?

Reply:
I thought that `modi_veros_run.sh` could be something like this:

```bash
source $CONDA_DIR/etc/profile.d/conda.sh

conda activate veros
if [ $? != 0 ]; then
    source modi_veros_install.sh

python $VEROS_SETUP
```

Then set `VEROS_SETUP` externally to allow it to run any setup file.

Reply:
Actually, if the conda versions on login and compute nodes are compatible, we could use this pattern:

```bash
conda create --prefix ~/modi_mount/conda-env-veros -y python=3.7
# activate using full path, should work on both login and compute node
conda activate ~/modi_mount/conda-env-veros
```

Then we could return to the original idea of doing all setup on the login node.

Reply:
Thanks! It works like a charm ;)